src/robotFactory.js

```diff
@@ -1,10 +1,82 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x, y }, chipVersion) {
```

Comment:
```suggestion
  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
```
Set default values while destructuring